![title](img/header.png)

<h1>Python - Dasar Tipe Data</h1>

Notebook ini digunakan untuk mempelajari dasar-dasar pemrograman Python. Tujuan dari Lab pertama ini adalah untuk memahami perintah-perintah dasar Python, mengenali tipe data dasar, dan melakukan operasi-operasi sederhana

<h2>Daftar Isi</h2>
<div class="alert alert-block alert-info" style="margin-top: 20px">
    <ul>
        <li>
            <a href="#hello">Hello world!</a>
            <ul>
                <li><a href="#version">Mengecek versi Python</a></li>
                <li><a href="#comments">Menambahkan komentar di Python</a></li>
                <li><a href="errors">Error di Python</a></li>
                <li><a href="python_error">Dapatkah Python mengenali error sebelum dijalankan?</a></li>
                <li><a href="exercise">Latihan: Membuat Program Pertama</a></li>
            </ul>
        </li>
        <li>
            <a href="#types_objects">Tipe dari objek di Python</a>
            <ul>
                <li><a href="int">Integers</a></li>
                <li><a href="float">Floats</a></li>
                <li><a href="convert">Mengubah tipe objek</a></li>
                <li><a href="bool">Tipe data Boolean</a></li>
                <li><a href="exer_type">Latihan: Tipe</a></li>
            </ul>
        </li>
        <li>
            <a href="#expressions">Ekspresi dan variabel</a>
            <ul>
                <li><a href="exp">Ekspresi</a></li>
                <li><a href="exer_exp">Latihan: Ekspresi</a></li>
                <li><a href="var">Variabel</a></li>
                <li><a href="exer_exp_var">Latihan: Ekspresi dan Variabel di Python</a></li>
            </ul>
        </li>
    </ul>
    <p>
        Perkiraan waktu mengerjakan: <strong>25 menit</strong>
    </p>
</div>

<hr>

<h2 name="hello">Hello World!</h2>

Sudah menjadi kebiasaan umum di kalangan programmer untuk menuliskan "hello world" ketika pertama kali mempelajari suatu bahasa pemrograman baru. Cell di bawah ini berisi perintah untuk menampilkan tulisan "hello world" di layar

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
[Tips]: Tekan  <kbd>Shift</kbd> + <kbd>Enter</kbd> untuk menjalankan kode pada cell di bawah ini.
</div>
<hr/>

In [1]:
# Coba jalankan cell ini

print('Hello, Python!')

Hello, Python!


Setelah menjalankan cell di atas, tulisan <code>Hello, Python!</code> seharusnya muncul di layar.

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Tips:] <code>print()</code> merupakan sebuah fungsi. Fungsi <code>print()</code> menerima masukan berupa string dan menampilkan di layar.    
</div>
<hr/>

<h3 id="version">Mengecek versi Python</h3>

<p>Terdapat dua versi Python yang banyak digunakan saat ini: Python 2 dan Python 3. Saat ini merupakan transisi dari Python 2 ke Python3, sehingga banyak library yang merencanakan untuk menghentikan dukungan terhadap Python 2.</p>
<p>Pada pelatihan ini versi Python yang digunakan adalah Python 3. Bagaimana cara kita mengecek versi Python yang digunakan? Terdapat beberapa cara untuk mengecek versi Python :
<ol>
    <li>Lihat bagian kanan atas pada Jupyter. Versi yang tercantum adalah Python 3.</li>
    <li>Menjalankan kode dalam bahasa Python seperti yang tercantum pada cell di bawah ini</li>
</ol>

</p>

In [2]:
# Mengecek versi Python yang terinstall

import sys
print(sys.version)

3.6.3 |Anaconda custom (64-bit)| (default, Oct 15 2017, 03:27:45) [MSC v.1900 64 bit (AMD64)]


<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Tips:] <code>sys</code> merupakan modul bawaan Python yang berisi berbagai parameter dan fungsi yang berhubungan dengan sistem Python. Sebelum digunakan, perlu dilakukan proses <code>import</code> terlebih dahulu.
</div>
<hr/>

<h3 id="comments">Menambahkan komentar di Python</h3>

<p>Dalam penulisan kode program (source code) seringkali perlu penambahan komentar pada kode. Komentar dapat digunakan untuk memberikan penjelasan tambahan mengenai kode sehingga dapat dipahami dengan mudah oleh orang lain ataupun sebagai referensi di masa mendatang.</p>

In [3]:
# Latihan menuliskan komentar

print('Hello, Python!') #Menampilkan tulisan di layar
# print('Hi')

Hello, Python!


<p>Setelah cell di atas dieksekusi, Anda melihat bahwa tulisan <code>Menampilkan tulisan di layar</code> tidak ditampilkan di layar karena tulisan tersebut merupakan komentar yang tidak dieksekusi oleh Python</p>

<p>Demikian juga dengan perintah <code>print('Hi')</code> yang diawali dengan tanda pagar (<code>#</code>) tidak akan dieksekusi oleh Python.</p>

<h3 id="errors">Errors di Python</h3>

<p>Error merupakan hal yang wajar terjadi dalam penulisan kode program. Apabila terjadi error, Python akan menampilkan pesan error. Pengenalan error sangat diperlukan untuk mengetahui lokasi dan penyebab error sehingga memudahkan Anda dalam memperbaiki error yang terjadi.</p>
<p>Sebagai contoh, jika Anda salah mengetik <code>print</code> dengan <code>frint</code>, maka Python akan menampilkan pesan error. Coba jalankan cell di bawah ini</p>

In [4]:
# Kesalahan pemanggilan fungsi print

frint("Hello, Python!")

NameError: name 'frint' is not defined

<p>Pesan error di atas menunjukkan :
    <ol>
        <li>Lokasi error (ditunjukkan dengan tanda panad)</li>
        <li>Jenis dari error yang terjadi</li>
    </ol>
</p>
<p>Pada contoh ini, Python mencoba menjalankan fungsi <code>frint</code>, tetapi tidak dapat menemukan fungsi tersebut karena fungsi <code>frint</code> tidak terdapat pada fungsi bawaan Python dan juga belum pernah dibuat oleh pengguna</p>

Apabila Anda melakukan jenis kesalahan yang berbeda, maka pesan error yang ditampilkan akan berbeda pula. Contoh di bawah ini merupakan kesalahan penulisan String tanpa tanda kutip dua di akhir String

In [ ]:
# Jalankan kode berikut

print("Hello, Python!)

<h3 id="python_error">Dapatkah Python mengenali error sebelum dijalankan?</h3>

<p>Python merupakan <i>interpreted language</i>, bukan <i>compiled language</i>. <i>Compiled language</i> akan memeriksa keseluruhan kode pada saat proses kompilasi, dan dapat memberikan peringatan sebelum kode dieksekusi. Sebaliknya, Python sebagai <i>interpreted language</i> langsung menjalankan kode program baris demi baris. Python akan berhenti menjalankan kode ketika mendapati adanya error, kecuali jika dilakukan error handling secara terpisah</p>

Jalankan cell di bawah ini dan amati hasilnya

In [ ]:
# Print string dan error untuk melihat urutan eksekusi

print("This will be printed")
frint("This will cause an error")
print("This will NOT be printed")

<h3 id="exercise">Latihan: Membuat Program Pertama</h3>

Tampilkan kata <code>Hello, world!</code> menggunakan fungsi <code>print()</code>

In [ ]:
# Tulis kode disini dan tekan Shift+Enter

Double-click __disini__ untuk menampilkan jawaban.

<!-- Your answer is below:

print("Hello, world!")

-->

<p>Now, let's enhance your code with a comment. In the code cell below, print out the phrase: <code>Hello, world!</code> and comment it with the phrase <code>Print the traditional hello world</code> all in one line of code.</p>

Tambahkan komentar pada kode program. Tampilkan tulisan <code>Hello, world!</code> dan komentar berupa kalimat <code>Tampilkan tulisan hello world</code> dalam satu baris

In [ ]:
# Tulis kode disini dan tekan Shift+Enter

Double-click __disini__ untuk menampilkan jawaban.

<!-- Your answer is below:

print("Hello, world!") # Print the traditional hello world

-->


<hr>

<h2 id="types_objects" align="center">Tipe dari objek di Python</h2>

Python merupakan bahasa pemrograman yang dapat menganut filosofi <i>object oriented programming</i>. Terdapat bermacam-macam tipe objek di Python. Tipe objek yang umum adalah : <i>string, integer, dan float</i>. Semua teks yang dituliskan pada Python merupakan <i>string</i>. Bilangan bulat (contoh: -1,0,100) memiliki tipe objek <i>integer</i> sedangkan bilangan pecahan (contoh : 3.14 , -37.5) memiliki tipe objek <i>float</i>.

<a align="center">
    <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/PY0101EN/Chapter%201/Images/TypesObjects.png" width="600">
</a>

<p>Cell di bawah ini berisi beberapa contoh.</p>

In [ ]:
# Integer

11

In [ ]:
# Float

2.14

In [ ]:
# String

"Hello, Python 101!"

Python dapat menampilkan tipe objek dari suatu variabel dengan menggunakan fungsi bawaan <code>type()</code> dengan parameter masukan berupa variabel yang akan dicek tipenya. Python akan menampilkan tipe objek integer dengan tulisan <code>int</code>, float dengan tulisan <code>float</code>, dan string dengan tulisan <code>string</code>.

In [ ]:
# Tipe dari angka 12

type(12)

In [ ]:
# Tipe dari angka 2.14

type(2.14)

In [ ]:
# Tipe dari string "Hello, Python 101!"

type("Hello, Python 101!")

<p>In the code cell below, use the <code>type()</code> function to check the object type of <code>12.0</code>.

Pada cell di bawah ini, gunakan fungsi <code>type()</code> untuk mengecek tipe dari angka <code>3.14</code>

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __disini__ untuk menampilkan jawaban.

<!-- Your answer is below:

type(3.14.0)

-->

<h3 id="int">Integers</h3>

Berikut merupakan contoh tipe objek integer. Integer berupa bilangan bulat, baik positif maupun negatif.

<a align="center">
    <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/PY0101EN/Chapter%201/Images/TypesInt.png" width="600">
</a>

Anda dapat mengecek tipe dari angka menggunakan fungsi <code>type()</code>

In [ ]:
# Tampilkan tipe dari angka -1

type(-1)

In [ ]:
# Tampilkan tipe dari angka 4

type(4)

In [ ]:
# Tampilkan tipe dari angka 0

type(0)

<h3 id="float">Floats</h3> 

<p>Float merupakan angka yang bernilai pecahan atau bilangan real. Komputer memiliki keterbatasan dalam memproses bilangan real, terutama mengenai berapa angka dibelakang koma yang dapat diproses. Nilai terbesar dan terkecil dari tipe <code>float</code> dapat dicek menggunakan perintah <code>sys.float_info</code>.</p>
<p>Pada cell di bawah ini, cek tipe objek dari masing-masing nilai yang tersedia.</p>

In [ ]:
# Tampilkan tipe dari angka 1.0

type(1.0) # Notice that 1 is an int, and 1.0 is a float

In [ ]:
# Tampilkan tipe dari angka 0.5

type(0.5)

In [ ]:
# Tampilkan tipe dari angka 0.56

type(0.56)

In [ ]:
# Tampilkan informasi mengenai tipe float

sys.float_info

<h3 id="convert">Mengubah tipe objek</h3>

<p>You can change the type of the object in Python; this is called typecasting. For example, you can convert an <i>integer</i> into a <i>float</i> (e.g. 2 to 2.0).</p>
<p>Let's try it:</p>

Pengubahan tipe objek dapat dilakukan menggunakan <i>typecasting</i>. Sebagai contoh, Anda dapat mengubah sebuah <i>integer</i> menjadi <i>float</i> (contoh : 2 menjadi 2.0)

In [ ]:
# Cek apakah angka di bawah ini merupakan integer

type(2)

<h4>Mengubah <i>integer</i> menjadi <i>float</i></h4>
<p>Ubahlah angka 2 menjadi float:</p>

In [ ]:
# Ubah angka 2 menjadi float

float(2)

In [ ]:
# Ubah angka 2 menjadi float dan cek  tipe objek

type(float(2))

Proses konversi integer menjadi float tidak akan mengubah nilai variabel. Tetapi, pengubahan float menjadi integer dapat menyebabkan perubahan nilai karena hilangnya nilai pecahan. Sebagai contoh, jika angka 1,1 diubah menjadi integer akan menghasilkan nilai 1 sehingga nilai pecahan sebesar 0,1 akan hilang

In [ ]:
# Pengubahan 1.1 dapat menghilangkan sebagian informasi

int(1.1)

<h4>Mengubah strings menjadi integer atau float</h4>

Pada beberapa kejadian, kita mendapati string yang berisi angka. Pada kasus ini, dapat dilakukan pengubahan string yang berisi angka menjadi integer dengan menggunakan fungsi <code>int()</code>

In [ ]:
# Ubah string menjadi integer

int('1')

<p>Jika Anda mengubah string yang berisi selain angka, maka Anda akan mendapati adanya error. Jalankan cell di bawah ini</p>



In [ ]:
# Mengubah string yang berisi kombinasi angka dan teks menjadi integer

int('1 or 2 people')


<p>String angka juga dapat diubah menjadi float dengan menggunakan fungsi <code>float()</code></p>

In [ ]:
# Mengubah string "1.2" menjadi float
float('1.2')

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Tip:] String dapat ditulis menggunakan tanda kutip tunggal(<code>'1.2'</code>) maupun kutip ganda (<code>"1.2"</code>), tetapi tidak boleh dicampur (e.g., <code>"1.2'</code>).
</div>
<hr/>

<h4>Mengubah angka menjadi string</h4>

<p>Integer juga dapat diubah menjadi <i>string</i> menggunakan fungsi <code>str()</code> </p>

In [ ]:
# Mengubah integer menjadi string

str(1)

<p>Dengan fungsi yang sama, float dapat pula diubah menjadi string</p> 

In [ ]:
# Mengubah float menjadi string

str(1.2)

<h3 id="bool">Tipe data boolean</h3>

<p><i>Boolean</i> merupakan tipe objek pada Python yang hanya memiliki dua nilai yaitu: <code>True</code> atau <code>False</code>:</p>

In [ ]:
# Nilai true

True

<p>Pastikan Anda menggunakan huruf kapital "T" pada True dan kapital "F" pada False</p>

In [ ]:
# Nilai false

False

<p>Pengecekan tipe objek boolean akan menampilkan <code>bool</code> yang merupakan singkatan dari <i>boolean</i>:</p> 



In [ ]:
# Tipe dari True

type(True)

In [ ]:
# Tipe dari False

type(False)

Objek boolean dapat diubah menjadi tipe objek lainnya. Jika boolean <code>True</code> diubah menjadi integer akan menghasilkan nilai satu (1), jika <code>False</code> diubah menjadi integer akan menghasilkan nol (0). Sebaliknya, jika integer 1 diubah menjadi Boolean akan menghasilkan <code>True</code> dan integer 0 jika diubah menjadi Boolean akan menghasilkan <code>False</code>

In [ ]:
# Mengubah True menjadi int

int(True)

In [ ]:
# Mengubah 1 menjadi boolean

bool(1)

In [ ]:
# Mengubah 0 menjadi boolean

bool(0)

In [ ]:
# Mengubah True menjadi float

float(True)

# <h3 id="exer_type">Latihan: Tipe</h3>

<p>Apakah tipe data dari operasi berikut: <code>6 / 2</code>?</p>

In [ ]:
# Tuliskan kode Anda di sini. Tekan Shift+Enter untuk mengeksekusi

Double-click __disini__ untuk melihat hasilnya.

<!-- Your answer is below:
type(6/2) # float
-->

<p>Apakah tipe data dari operasi berikut: <code>6 // 2</code>? (Perhatikan penggunaan operator <code>//</code>.)</p>

In [ ]:
# Tuliskan kode Anda di sini. Tekan Shift+Enter untuk mengeksekusi

Double-click __disini__ untuk melihat hasilnya.

<!-- Your answer is below:
type(6//2) # int, as the double slashes stand for integer division 
-->

<hr>

<h2 id="expressions">Ekspresi dan **Variabel**</h2>

<h3 id="exp">Ekspresi</h3>


<p>Ekspresi pada Python dapat mengandung beberapa operasi yang sesuai pada tipe data tertentu (contoh: penjumlahan pada integer dan float)</p>

In [5]:
# Ekspresi operasi penjumlahan

43 + 60 + 16 + 41

160

<p>Operasi pengurangan dapat dilakukan menggunakan operator minus. Pada contoh ini, angka yang dihasilkan bernilai minus:</p>

In [6]:
# Ekspresi operasi pengurangan

50 - 60

-10

<p>Operasi perkalian dapat menggunakan operator asteriks:</p>


In [7]:
# Ekspresi operasi perkalian

5 * 5

25

<p>Operasi pembagian dapat menggunakan operator forward slash:</p>


In [8]:
# Operasi pembagian

25 / 5

5.0

In [9]:
# Operasi pembagian

25 / 6

4.166666666666667

<p>Pada latihan di atas,  double slash digunakan untuk pembagian integer, dimana hasilnya akan dibulatkan ke integer terdekat</p>

In [10]:
# Operasi pembagian integer

25 // 5

5

In [11]:
# Operasi pembagian integer

25 // 6

4



<h3 id="exer_exp">Latihan: Ekspresi</h3>

  
 <p>Tuliskan sebuah ekspresi yang mengubah 160 menit menjadi jam</p>

In [12]:
# Tuliskan kode Anda di sini. Tekan Shift+Enter untuk mengeksekusi

Double-click __disini__ untuk melihat hasilnya.

<!-- Your answer is below:
160/60 
# Or 
160//60
-->

<p>Python melakukan eksekusi operasi matematika sesuai kaidah matematika yang benar. Pada contoh di bawah ini, Python akan melakukan perkalian terlebih dahulu dan diikuti penjumlahan</p>

In [13]:
# Ekspresi matematika

30 + 2 * 60

150


<p>Penggunaan kurung menandakan prioritas operasi matematika, sehingga penjumlahan akan dilakukan terlebih dahulu sebelum perkalian</p>

In [14]:
# Ekspresi matematika

(30 + 2) * 60

1920

<h3 id="var">Variabel</h3>

<p>Sama seperti bahasa pemrograman lainnya, <i>variabel</i> digunakan untuk menyimpan nilai sehingga dapat digunakan pada tahap selanjutnya. Sebagai contoh: </p>

In [15]:
# Simpan hasil kalkulasi ke dalam sebuah variabel

x = 43 + 60 + 16 + 41



<p>Nilai dari variabel <code>x</code> dapat ditamplkan dengan mengetik nama variabel</p>

In [16]:
# Tampilkan nilai variabel

x

160


<p>Anda juga dapat melakukan operasi pada variabel <code>x</code> dan menyimpan hasilnya pada variabel lainnya</p>


In [17]:
# Gunakan variabel lain untuk menyimpan nilai dari operasi berikut

y = x / 60
y

2.6666666666666665

<p>Jika suatu nilai disimpan pada sebuah variabel yang sudah ada, maka nilai baru akan menggantikan nilai yang lama</p>

In [18]:
# Ganti nilai variabel dengan nilai baru

x = x / 60
x

2.6666666666666665

<p>Pemberian nama variabel sebaiknya menggunakan nama yang memiliki arti jelas sehingga kode sumber dapat dipahami dengan mudah</p>

In [19]:
# Berikan nama variabel dengan arti yang jelas

total_min = 43 + 42 + 57 #Total durasi lagu dalam sebuah album dalam satuam menit
total_min

142

In [20]:
# Berikan nama variabel dengan arti yang jelas

total_hours = total_min / 60 # Total durasi dalam satuan jam
total_hours

2.3666666666666667

<p> Pada cell di atas, durasi dari 3 lagu dijumlahkan (dalam satuan menit) dan disimpan di <code>total_min</code>. Selanjutnya, dilakukan pembagian dengan 60 untuk mendapatkan total durasi dalam satuan jam dan disimpan di variabel <code>total_hours</code>. Sebagai alternatif, semua operasi tersebut dapat dilakukan dalam satu ekspresi seperti di bawah ini : </p>

In [21]:
# Complicate expression

total_hours = (43 + 42 + 57) / 60  # Total hours in a single expression
total_hours

2.3666666666666667

<p>Jika Anda menginginkan total jam dalam tipe integer, Anda dapat mengubah pembagian float menjadi pembagian integer  (menggunakan operasi  <code>//</code>)</p>

<h3 id="exer_exp_var">Latihan: Ekspresi dan Variabel di Python</h3>

<p>Berapakah nilai  <code>x</code> dimana <code>x = 3 + 2 * 2</code></p>

In [22]:
# Tuliskan kode Anda di sini. Tekan Shift+Enter untuk mengeksekusi

Double-click __disini__ untuk melihat hasilnya.

<!-- Your answer is below:
7
-->



<p>Berapa nilai <code>y</code> dimana <code>y = (3 + 2) * 2</code>?</p>

In [23]:
# Tuliskan kode Anda di sini. Tekan Shift+Enter untuk mengeksekusi

Double-click __disini__ untuk melihat hasilnya.

<!-- Your answer is below:
10
-->


<p>Berapa nilai dari <code>z</code> dimana <code>z = x + y</code>?</p>

In [24]:
# Tuliskan kode Anda di sini. Tekan Shift+Enter untuk mengeksekusi

Double-click __disini__ untuk melihat hasilnya.
<!-- Your answer is below:
17
-->

<h3>About the Authors:</h3>  
<p><a href="https://www.linkedin.com/in/joseph-s-50398b136/" target="_blank">Joseph Santarcangelo</a> is a Data Scientist at IBM, and holds a PhD in Electrical Engineering. His research focused on using Machine Learning, Signal Processing, and Computer Vision to determine how videos impact human cognition. Joseph has been working for IBM since he completed his PhD.</p>

Other contributors: <a href="www.linkedin.com/in/jiahui-mavis-zhou-a4537814a">Mavis Zhou</a>

<hr>
<p>Copyright &copy; 2018 IBM Developer Skills Network. This notebook and its source code are released under the terms of the <a href="https://cognitiveclass.ai/mit-license/">MIT License</a>.</p>